core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java

```diff
@@ -131,11 +131,20 @@ public Schema createSchema(NameIdentifier ident, String comment, Map<String, Str
     Map<String, String> updatedProperties =
         StringIdentifier.newPropertiesWithId(stringId, properties);
 
-    doWithCatalog(
-        catalogIdent,
-        c -> c.doWithSchemaOps(s -> s.createSchema(ident, comment, updatedProperties)),
-        NoSuchCatalogException.class,
-        SchemaAlreadyExistsException.class);
+    Schema createdSchema =
+        doWithCatalog(
+            catalogIdent,
+            c -> c.doWithSchemaOps(s -> s.createSchema(ident, comment, updatedProperties)),
+            NoSuchCatalogException.class,
+            SchemaAlreadyExistsException.class);
+
+    // If the Schema is maintained by the Gravitino's store, we don't have to store again.
+    boolean isManagedSchema = isManagedSchema(createdSchema);
+    if (isManagedSchema) {
```

Comment:
Can these code be split into a common method? Looks like there are multiple usage with little change.